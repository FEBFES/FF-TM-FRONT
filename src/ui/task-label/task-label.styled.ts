import styled, { RuleSet, css } from 'styled-components';
import { ITaskLabelType } from './task-label';

export type StyledVariants<E extends string> = {
  [key in E]?: RuleSet;
};

const taskTypes: StyledVariants<ITaskLabelType> = {
  QUESTION: css`
    background: var(--task-label-q-bg);
    border: 1px solid var(--task-label-q-border);
    color: var(--task-label-q-color);
  `,
  RESEARCH: css`
    background: var(--task-label-r-bg);
    border: 1px solid var(--task-label-r-border);
    color: var(--task-label-r-color);
  `,
  BUG: css`
    background: var(--task-label-b-bg);
    border: 1px solid var(--task-label-b-border);
    color: var(--task-label-b-color);
  `,
  FEATURE: css`
    background: var(--task-label-f-bg);
    border: 1px solid var(--task-label-f-border);
    color: var(--task-label-f-color);
  `,
};

export const STaskLabel = styled.div<{ type: ITaskLabelType }>`
  ${({ type }) => taskTypes[type]};
  padding: 2px 6px;
  border-radius: 6px;
  font-size: 10px;
  line-height: 14px;
  font-weight: 400;
`;
