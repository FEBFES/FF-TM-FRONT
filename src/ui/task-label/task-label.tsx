import React from 'react';
import { STaskLabel } from './task-label.styled';

export type ITaskLabelType =
  | 'NONE'
  | 'QUESTION'
  | 'RESEARCH'
  | 'BUG'
  | 'FEATURE';

export interface TaskLabelProps {
  type: ITaskLabelType | null;
}

export const TaskLabel: React.FC<TaskLabelProps> = ({ type }): JSX.Element => {
  return (
    <>
      {type !== 'NONE' && type && (
        <STaskLabel type={type}>{type.toLowerCase()}</STaskLabel>
      )}
    </>
  );
};
