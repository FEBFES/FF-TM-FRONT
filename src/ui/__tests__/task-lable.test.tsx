import React from 'react';
import {TaskLabel} from '../task-label/task-label';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
    it('SNAPSHOTS: UI - TaskLabel (null) component - render', () => {
        const elem = renderer.create(<TaskLabel type={null} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
    it('SNAPSHOTS: UI - TaskLabel (NONE) component - render', () => {
        const elem = renderer.create(<TaskLabel type={'NONE'} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
    it('SNAPSHOTS: UI - TaskLabel (QUESTION) component - render', () => {
        const elem = renderer.create(<TaskLabel type={'QUESTION'} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
    it('SNAPSHOTS: UI - TaskLabel (RESEARCH) component - render', () => {
        const elem = renderer.create(<TaskLabel type={'RESEARCH'} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
    it('SNAPSHOTS: UI - TaskLabel (BUG) component - render', () => {
        const elem = renderer.create(<TaskLabel type={'BUG'} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
    it('SNAPSHOTS: UI - TaskLabel (FEATURE) component - render', () => {
        const elem = renderer.create(<TaskLabel type={'FEATURE'} />).toJSON();
        expect(elem).toMatchSnapshot();
    });
})