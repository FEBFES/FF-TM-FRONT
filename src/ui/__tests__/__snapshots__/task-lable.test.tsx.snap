// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (BUG) component - render 1`] = `
<div
  className="sc-bcPKhP bmaxfA"
  type="BUG"
>
  bug
</div>
`;

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (FEATURE) component - render 1`] = `
<div
  className="sc-bcPKhP kTnWAc"
  type="FEATURE"
>
  feature
</div>
`;

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (NONE) component - render 1`] = `null`;

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (QUESTION) component - render 1`] = `
<div
  className="sc-bcPKhP imiGFX"
  type="QUESTION"
>
  question
</div>
`;

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (RESEARCH) component - render 1`] = `
<div
  className="sc-bcPKhP iCLXls"
  type="RESEARCH"
>
  research
</div>
`;

exports[`<Button /> SNAPSHOTS: UI - TaskLabel (null) component - render 1`] = `null`;
